app/battle_prepare/battle_prepare.py

```diff
@@ -0,0 +1,21 @@
+from app.constants.lists import PROPERTY_LIST
+
+
+def battle_prepare(knight: dict) -> None:
+    def add_potion_effect(prop: str) -> None:
+        potion_effect = knight.get("potion", {}).get("effect")

```

Comment:
It is always true due to:
`if "potion" in knight and knight["potion"] is not None:`

Reply:
Do you mean that "if prop in potion_effect:" can be removed? But it is impossible for this case, since not all the effects from PROPERTY_LIST are present in KNIGHTS data. red_knight["potion"] does not have "protection" effect.